src/scrcmd_c.c

```diff
@@ -1725,7 +1725,7 @@ BOOL ScrCmd_GetPhoneBookRematch(SCRIPTCONTEXT *ctx) {
     u16 *r6 = ScriptGetVarPointer(ctx);
     struct PhoneBook *phoneBook = AllocAndReadPhoneBook(32);
     HandleLoadOverlay(FS_OVERLAY_ID(OVY_26), OVY_LOAD_ASYNC);
-    *r6 = PhoneBookTrainerGetRematchInfo(r4, ctx->fsys->savedata, phoneBook, Field_GetTimeOfDayWildParam(ctx->fsys));
+    *r6 = PhoneBookTrainerGetRematchInfo(r4, ctx->fsys->savedata, phoneBook, (TimeOfDayWildParam)(u8)Field_GetTimeOfDayWildParam(ctx->fsys));
```

Comment:
Why the double casting here? I'm guessing it's cause of the pedantic warnings, but those were added a few days ago.

Reply:
It's required for matching, I think. If I don't cast to u8 before, the compiler considers it to be signed, and it executes an arithmetic right shift (asrs) instead of logical (lsrs). The other option was not to use the enum, but I'm quite an advocate for using them whenever possible